### PR TITLE
Updated tests

### PR DESCRIPTION
**Pull request by holailusoria**

### CODE REVIEW

File: Projects/GeometryTestProject/TestRelation.cs [hunk 1]
```diff
@@ -49,8 +62,30 @@ public void TestRelationPageSizeNULL()
     {
       DataQueryBuilder qb = DataQueryBuilder.Create();
       qb.AddAllProperties();
-      qb.SetRelationsPageSize( null );
+      qb.SetRelationsPageSize( 0 );
+      qb.SetRelated( new List<String> { "Related" } );
+      IList<Dictionary<String, Object>> res = Backendless.Data.Of( "Order" ).Find( qb );
```
Comment (markpiller) [inline Projects/GeometryTestProject/TestRelation.cs:67]: what does this test test? I do not see any checks

File: Projects/GeometryTestProject/TestRelation.cs [hunk 2]
```diff
@@ -49,8 +62,30 @@ public void TestRelationPageSizeNULL()
...
+
+    [TestMethod]
+    public void TestSort()
+    {
+      DataQueryBuilder qb = DataQueryBuilder.Create();
+      qb.AddAllProperties();
+      qb.SetRelated( new List<String> { "Related" } );
+      qb.SetSortBy( new List<String> { "age" } );
       IList<Dictionary<String, Object>> res = Backendless.Data.Of( "Order" ).Find( qb );
+      Assert.IsTrue( (Double) res[ 0 ][ "age" ] == 5.0 && (Double) res[1]["age"] == 10.0 );
```
Comment (markpiller) [inline Projects/GeometryTestProject/TestRelation.cs:88]: split this into two separate Assert statements

File: Projects/GeometryTestProject/TestRelation.cs [hunk 3]
```diff
@@ -24,33 +37,66 @@ public void TestRelationDepth()
...
+      Object[] rel = (Object[]) res[ 0 ][ "Related" ];
 
-      Assert.IsFalse( res[ 0 ].ContainsKey( "Related" ) );
-      Assert.IsFalse( res[ 1 ].ContainsKey( "Related" ) );
+      Assert.IsTrue( rel.Length == 0 );
     }
 
     [TestMethod]
-    public void TestRelationPageSizeNULL()
+    public void TestRelationsDepth()
```
Comment (markpiller) [inline Projects/GeometryTestProject/TestRelation.cs:79]: this method does not test relations depth

File: Projects/GeometryTestProject/TestRelation.cs [hunk 4]
```diff
@@ -24,33 +37,119 @@ public void TestRelationDepth()
...
+    }
+
+    [TestMethod]
+    public void TestRelationsDepth()
+      DataQueryBuilder qb = DataQueryBuilder.Create();
+      qb.AddAllProperties();
+      qb.SetRelationsDepth( 1 );
+      qb.SetRelated( new List<String> { "Country", "City" } );
+      IList<Dictionary<String, Object>> res = Backendless.Data.Of( "CountryLanguage" ).Find( qb );
+      Object entry =  res[0]["Country"];
```
Comment (markpiller) [inline Projects/GeometryTestProject/TestRelation.cs:130]: why not check if "Country" is present with Assert ?

File: Projects/GeometryTestProject/TestRelation.cs [hunk 5]
```diff
@@ -24,33 +37,119 @@ public void TestRelationDepth()
...
+      IList<Dictionary<String, Object>> res = Backendless.Data.Of( "CountryLanguage" ).Find( qb );
+      foreach( Dictionary<String, Object> entry in res )
+      {
+        if( i == 0 )
+        {
+          gg = entry[ "Percentage" ];
+          i++;
+          continue;
+        }
+        Assert.IsTrue( (Double) gg < (Double) entry[ "Percentage" ] );
```
Comment (markpiller) [inline Projects/GeometryTestProject/TestRelation.cs:150]: we do not need to check if groupby works properly. What you need to ensure is that you actually got expected data

File: Projects/TestProject/ExcludePropertiesTests.cs [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+{
+  [TestClass]
+  public class ExcludePropertiesTests
+  {
+    [ClassInitialize]
+    public static void TestExcludeSetupData( TestContext context )
+    {
+      try
+      {
+        Backendless.Data.Describe( "A" );
```
Comment (markpiller) [inline Projects/TestProject/ExcludePropertiesTests.cs:18]: use something more meaningful than A and B

File: Projects/TestProject/ExcludePropertiesTests.cs [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        data.Add( "adress", "Tom Street" );
+
+        Dictionary<String, Object> dataIdChildren_2 = Backendless.Data.Of( "B" ).Save( data );//Second object int the "B" table
+        ////////////////////////////////////////////////////////////////////////////////////////
+
+        ///Сreating a connection between the objects "Order" and "Area"///
+
+        Object[] children = new Object[] { dataIdChildren_1 };
+
+        Backendless.Data.Of( "A" ).SetRelation( dataIdParent_1, "table_B:B:n", children );//First relation
```
Comment (markpiller) [inline Projects/TestProject/ExcludePropertiesTests.cs:57]: change the name of the column to make it more meaningful

File: Projects/TestProject/GeometryTests.cs [hunk 8]
```diff
@@ -5,12 +5,147 @@
...
+        CreateColumns( GEOMETRY_NAME, "GeoValue" );
+
+        data.Add( "GeoValue", new Point().SetX( 10.2 ).SetY( 48.5 ) );
+
+
+        Backendless.Data.Of( "GeoData" ).Save( data );
+      }
+    }
+
+    static String CreateToken()
```
Comment (markpiller) [inline Projects/TestProject/GeometryTests.cs:93]: rename to LoginAndGetToken()

File: Projects/TestProject/GeometryTests.cs [hunk 9]
```diff
@@ -5,12 +5,147 @@
...
+      HttpRequestMessage request = new HttpRequestMessage( HttpMethod.Post, "https://develop.backendless.com/console/home/login" );
+      request.Content = new StringContent( "{\"login\":\"nikita@themidnightcoders.com\",\"password\":\"Holailusoria1411\"}", Encoding.UTF8, "application/json" );
+
+      return client.SendAsync( request ).GetAwaiter().GetResult().Headers.GetValues( "auth-key" ).ToArray()[ 0 ];
+    }
+
+    static void CreateColumns( String typeName, String columnName, bool TableIsCreated = true )
+    {
+      client = new HttpClient();
+      client.BaseAddress = new Uri( "https://develop.backendless.com" );
```
Comment (markpiller) [inline Projects/TestProject/GeometryTests.cs:104]: move URL to a constant

File: Projects/TestProject/GeometryTests.cs [hunk 10]
```diff
@@ -5,12 +5,147 @@
...
+
+    static String CreateToken()
+    {
+      HttpRequestMessage request = new HttpRequestMessage( HttpMethod.Post, "https://develop.backendless.com/console/home/login" );
+      request.Content = new StringContent( "{\"login\":\"nikita@themidnightcoders.com\",\"password\":\"Holailusoria1411\"}", Encoding.UTF8, "application/json" );
+
+      return client.SendAsync( request ).GetAwaiter().GetResult().Headers.GetValues( "auth-key" ).ToArray()[ 0 ];
+    }
+
+    static void CreateColumns( String typeName, String columnName, bool TableIsCreated = true )
```
Comment (markpiller) [inline Projects/TestProject/GeometryTests.cs:101]: method creates one column, rename to reflect that (CreateColumn)

File: Projects/TestProject/GeometryTests.cs [hunk 11]
```diff
@@ -237,7 +379,7 @@ public void LineStringWKTEquals()
       Assert.AreEqual( line.AsWKT(), StrCoordinates, "LineString WKT data are not equals" );
 
       pers.Add( "LineValue", line );
-      Backendless.Data.Of( "Person" ).Save( pers );
+      Backendless.Data.Of( "GeoData" ).Save( pers );
```
Comment (markpiller) [inline Projects/TestProject/GeometryTests.cs:382]: Why?  What is this for?

File: Projects/TestProject/GeometryTests.cs [hunk 12]
```diff
@@ -254,27 +396,7 @@ public void PolygonWKTEquals()
...
-      Polygon poly = (Polygon) wkt.Read( "POLYGON((-77.05786152 38.87261877,-77.0546978 38.87296123,-77.05317431 38.87061405," +
-      "-77.0555883 38.86882611,-77.05847435 38.87002898,-77.05786152 38.87261877),(-77.05579215 38.87026286," +
-      "-77.05491238 38.87087264,-77.05544882 38.87170794,-77.05669337 38.87156594,-77.05684357 38.87072228," +
-      "-77.05579215 38.87026286))" );
-
-      Assert.AreEqual( pers[ "PointCol" ], point, "Point WKT data are not equals" );
-      Assert.AreEqual( pers[ "GeometryCol" ], geometry, "Geometry(Point) WKT data are not equals" );
-      Assert.AreEqual( pers[ "LineStringCol" ], line, "LineString WKT data are not equals" );
-      Assert.AreEqual( pers[ "PolygonCol" ], poly, "Polygon WKT data are not equals" );
+      Backendless.Data.Of( "GeoData" ).Save( pers );
```
Comment (markpiller) [inline Projects/TestProject/GeometryTests.cs:399]: why?

File: Projects/TestProject/GeometryTests.cs [hunk 13]
```diff
@@ -5,12 +5,146 @@
...
+
+
+        Backendless.Data.Of( "GeoData" ).Save( data );
+      }
+    }
+
+    static String LoginAndGetToken()
+    {
+      HttpRequestMessage request = new HttpRequestMessage( HttpMethod.Post, "https://develop.backendless.com/console/home/login" );
+      request.Content = new StringContent( "{\"login\":\"nikita@themidnightcoders.com\",\"password\":\"Holailusoria1411\"}", Encoding.UTF8, "application/json" );
```
Comment (markpiller) [inline Projects/TestProject/GeometryTests.cs:96]: not a good idea to commit your password to github

File: Projects/TestProject/RelationsTests.cs [hunk 14]
```diff
@@ -0,0 +1,299 @@
...
+      DataQueryBuilder qb = DataQueryBuilder.Create();
+      qb.AddAllProperties();
+      qb.SetRelationsDepth( 1 );
+      qb.SetRelated( new List<String> { "Country", "Capital" } );
+      IList<Dictionary<String, Object>> res = Backendless.Data.Of( "CountryLanguage" ).Find( qb );
+
+      if(res[ 0 ].ContainsKey( "Country" ) )
+      {
+        Object entry = res[ 0 ][ "Country" ];
+        Assert.IsFalse( ( (Dictionary<Object, Object>) entry ).ContainsKey( "Capital" ) );
```
Comment (markpiller) [inline Projects/TestProject/RelationsTests.cs:282]: What are you testing here? If object has Country and doesn't have Capital, the test passes? Why are you requesting `{ "Country", "Capital" }` ? Is "Capital" a relation column in "CountryLanguage" ?

Comment (holailusoria) [reply Projects/TestProject/RelationsTests.cs:282]: If object has Country and doesn't have Capital, the test passes? - Yes.

Why are you requesting { "Country", "Capital" } ? - This is error, i fixed it.

Is "Capital" a relation column in "CountryLanguage" ? - No, "Capiral", - a relation column in "Country".

What are you testing here? - I testing "relationsDepth field". When I changed the data type, I could break something, this test checks if the expected value of the "RelationsDepth" matches the real

**Pull request merged**